src/OrleansServiceBus/Providers/Streams/EventHub/ICheckpointSettings.cs

```diff
@@ -0,0 +1,99 @@
+﻿
+using System;
+using System.Collections.Generic;
+using Orleans.Providers;
+
+namespace Orleans.ServiceBus.Providers
+{
+    public interface ICheckpointSettings
+    {
+        /// <summary>
+        /// Azure table storage data connections string
+        /// </summary>
+        string DataConnectionString { get; }
+
+        /// <summary>
+        /// Azure storage table name where the checkpoints will be stored
+        /// </summary>
+        string TableName { get; }
+
+        /// <summary>
+        /// How often to persist the checkpoints, if they've changed.
+        /// </summary>
+        TimeSpan PersistCadence { get; }
+
+        string DeploymentId { get; }
```

Comment:
It's intended to allow the partitioning of data by service, so deployment Id is not a good name, in retrospect. It is comparable to the event hub consumer group, but does not require the creation of a new event hub consumer group to change. It would be nice to use the consumer group for this, but that would complicate testing, as we'd constantly need to create and destroy event hub consumer groups, which is very slow (undesirable).

ServiceName? ServiceId? ...?